src/v/kafka/server/snc_quota_manager.cc

```diff
@@ -59,6 +61,71 @@ to_soa_layout(const std::vector<ingress_egress_state<T>>& v) {
 
 namespace {
 
+template<class T>
+void add_ingress_egress_gauges(
+  std::vector<ss::metrics::metric_definition>& metric_defs,
+  ss::metrics::metric_name_type name,
+  T&& value_getter,
+  ss::metrics::description desc,
+  const std::vector<ss::metrics::label>& aggregate_labels) {
+    static const auto direction_label = ss::metrics::label("direction");
+    static const auto label_ingress = direction_label("ingress");
+    static const auto label_egress = direction_label("egress");
+    metric_defs.emplace_back(ss::metrics::make_gauge(
+                               name,
+                               [value_getter] { return value_getter().in; },
+                               desc,
+                               {label_ingress})
+                               .aggregate(aggregate_labels));
+    metric_defs.emplace_back(ss::metrics::make_gauge(
+                               name,
+                               [value_getter] { return value_getter().eg; },
+                               desc,
+                               {label_egress})
+                               .aggregate(aggregate_labels));
+}
+
+} // namespace
+
+void snc_quotas_probe::setup_metrics() {
+    namespace sm = ss::metrics;
+
+    if (config::shard_local_cfg().disable_metrics()) {
+        return;
+    }
+
+    const auto aggregate_labels = config::shard_local_cfg().aggregate_metrics()
+                                    ? std::vector<sm::label>{sm::shard_label}
+                                    : std::vector<sm::label>{};
+    std::vector<ss::metrics::metric_definition> metric_defs;
+
+    if (ss::this_shard_id() == quota_balancer_shard) {
+        metric_defs.emplace_back(sm::make_counter(
+          "balancer_runs",
+          [this] { return _balancer_runs; },
+          sm::description(
+            "{}: Number of times throughput quota balancer has been run")));
+    }
+    add_ingress_egress_gauges(
+      metric_defs,
+      "quota_effective",
+      [this] { return _qm.get_quota(); },
+      sm::description("{}: Currently effective quota, in bytes/s"),
+      aggregate_labels);
+    add_ingress_egress_gauges(
+      metric_defs,
+      "throughput",
+      [this] { return _qm.get_throughput(); },
+      sm::description("{}: Throughput measurement at the time of last "
```

Comment:
> "{}: Thr

is this a format string that's filled in automatically?

Reply:
Nope, that's an artifact of copypasting from another metrics code.
```
# HELP vectorized_kafka_quotas_snc_throughput {}: Throughput measurement at the time of last balancer run, in bytes/s
```

Reply:
ahh, gotcha.